lib/DynamoDBContextStore.js

```diff
@@ -0,0 +1,259 @@
+'use strict';
+const AWS = require('aws-sdk');
+const primaryKey = Symbol("private");
+const createTableIfNecessary = Symbol("private");
+
+module.exports = class DynamoDBContextStore {
+
+	/**
+	 * @typedef {Object} TableOptions
+	 * @prop {String=} name The name of the table
+	 * @prop {String=} hashKey The name of the table
+	 * @prop {String=} prefix String pre-pended to the app ID and used for the hashKey
+	 * @prop {String|Object=} sortKey Optional sort key definition
+	 * @prop {Number=} readCapacityUnits Number of consistent reads per second. Used only when table is created
+	 * @prop {Number=} writeCapacityUnits Number of writes per second. Used only when table is created
+	 */
+	/**
+	 * @typedef {Object} DynamoDBContextStoreOptions
+	 * @prop {TableOptions=} table The table options
+	 */
+	/**
+	 * Create a context store instance instance
+	 * @param {DynamoDBContextStoreOptions} [options] Optionally, pass in a configuration object
+	 */
+	constructor(options = {}) {
+		this.table = {
+			name: "smartapp",
+			hashKey: 'id',
+			sortKey: null,
+			prefix: 'ctx:'
+		};
+
+		if (options.table) {
+			const table = options.table;
+			this.table.name = table.name || this.table.name;
+			this.table.hashKey = table.hashKey || this.table.hashKey;
+			this.table.prefix = table.prefix || this.table.prefix;
+			if (typeof options.table.sortKey === 'string') {
+				this.table.sortKey = {
+					AttributeName: options.table.sortKey,
+					AttributeType: 'S',
+					AttributeValue: 'context',
+					KeyType: 'RANGE'
+				}
+			} else {
+				this.table.sortKey = options.table.sortKey
+			}
+		}
+
+		if (options.client) {
+			this.client = options.client;
+		}
+		else {
+			if (options.AWSConfigPath) {
+				AWS.config.loadFromPath(options.AWSConfigPath);
+			}
+			else if (options.AWSConfigJSON) {
+				AWS.config.update(options.AWSConfigJSON);
+			}
+			else {
+				this.AWSRegion = options.AWSRegion || 'us-east-1';
+				AWS.config.update({region: this.AWSRegion});
+			}
+			this.client = new AWS.DynamoDB();
+		}
+
+		if (options.autoCreate !== false) {
+			this[createTableIfNecessary](options);
+		}
+	}
+
+	get(installedAppId) {
+		let params = {
+			TableName: this.table.name,
+			Key: {
+				[this.table.hashKey]: {S: this[primaryKey](installedAppId)}
+			},
+			ConsistentRead: true
+		};
+
+		if (this.table.sortKey) {
+			params.Key[this.table.sortKey.AttributeName] = {[this.table.sortKey.AttributeType]: this.table.sortKey.AttributeValue}
+		}
+
+		return new Promise((resolve, reject) => {
+			this.client.getItem(params, function(err, data) {
+				if (err) {
+					reject(err);
+				} else {
+					if (data && data.Item) {
+						const item = data.Item
+						resolve({
+							installedAppId: AWS.DynamoDB.Converter.output(item.installedAppId),
+							locationId: AWS.DynamoDB.Converter.output(item.locationId),
+							authToken: AWS.DynamoDB.Converter.output(item.authToken),
+							refreshToken: AWS.DynamoDB.Converter.output(item.authToken),
```

Comment:
This line appears to incorrectly set `refreshToken` to `authToken`.

Reply:
:thumbsup: